### PR TITLE
Disable version mismatch warning when doing tab autocompletion

### PR DESCRIPTION
**Pull request by jimmykarily**
The warning was always printed and was messing up the autocompletion. Now we explicitly disable the warning when doing autocompletion. There is still a chance that an out-of-sync client can fail to do the autocompletion but it's better to let it try than break completely.

### CODE REVIEW

File: internal/cli/usercmd/client.go [hunk 1]
```diff
@@ -99,6 +99,9 @@ type APIClient interface {
 	ChartList() ([]models.AppChart, error)
 	ChartShow(name string) (models.AppChart, error)
 	ChartMatch(prefix string) (models.ChartMatchResponse, error)
+
+	DisableVersionWarning()
+	VersionWarningEnabled() bool
```
Comment (jimmykarily) [inline internal/cli/usercmd/client.go:104]: I need to update the "fake" too. On it.

Comment (jimmykarily) [reply internal/cli/usercmd/client.go:104]: done

File: internal/cli/commons.go [hunk 2]
```diff
@@ -56,6 +56,7 @@ func matchingAppsFinder(cmd *cobra.Command, args []string, toComplete string) ([
 		return nil, cobra.ShellCompDirectiveNoFileComp
 	}
 
+	app.API.DisableVersionWarning()
```
Comment (andreas-kupries) [inline internal/cli/commons.go:59]: There are lots of `ValidArgsFunction`s not declared by name, but as a anon function, i.e. `ValidArgsFunction: func(cmd *cobra.Command ... ) { ... }, ...`, directly in the Cmd structures

All of these seem to be missing. `commons.go` only contains the matchers used multiple times.

**Review by andreas-kupries — CHANGES_REQUESTED** [review]
Lots more matchers to fix than done (the anon ones).

File: internal/cli/commons.go (hunk 2 above)
Comment (jimmykarily) [reply internal/cli/commons.go:59]: Good catch! Fixed here: 8e5b5111

File: pkg/api/core/v1/client/http.go [hunk 3]
```diff
@@ -36,6 +36,18 @@ func wrapResponseError(err error, code int) *responseError {
...
+func (c *Client) DisableVersionWarning() {
+	value := false
+	c.versionWarning = &value
+}
+
+// VersionWarningEnabled returns true if versionWarning field is either not
+// set of true. That makes "true" the default value unless DisableVersionWarning
+// is called.
+func (c *Client) VersionWarningEnabled() bool {
+	return c.versionWarning == nil || *c.versionWarning
```
Comment (andreas-kupries) [inline pkg/api/core/v1/client/http.go:48]: Name the field `noVersionWarning`, with a default of `false`. `Disable...` then simply sets it to `true`.
That gets rid of all the gymnastics with a pointer. Just a plain bool.

**Review by andreas-kupries — APPROVED** [review]
Approved, with a comment regarding the implementation of the flag.

File: pkg/api/core/v1/client/http.go (hunk 3 above)
Comment (jimmykarily) [reply pkg/api/core/v1/client/http.go:48]: I'm not sure which one I dislike more, double negations or the pointer hacks. I'll try your suggestion.

**Review by andreas-kupries — APPROVED** [review]

**Comment by thehejik** [comment]
I was unable to reproduce this issue but I guess I'd need to build epinio CLI from the commit just before merging this PR. Previous `epinio-1.3.0` CLI doesn't contain the warning yet for the version mismatch:
```
⚠️  Epinio server version (v1.3.0) doesn't match the client version (v1.4.0)
```

The autocomplete script works as expected in epinio release 1.4.0 under `bash`.
```
$ source <(epinio completion bash)
$ epinio c<TAB><TAB>
client-sync    completion     configuration
```

**Pull request merged**